pkg/apiserver/webhook/configmap_webhook.go

```diff
@@ -0,0 +1,121 @@
+	corev1 "k8s.io/api/core/v1"
+	controllerclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	controllers "antrea.io/nephe/pkg/controllers/cloud"
+	"antrea.io/nephe/pkg/controllers/config"
+)
+
+// ConfigMapValidator is used to validate ConfigMap.
+type ConfigMapValidator struct {
+	Client       controllerclient.Client
+	Log          logr.Logger
+	decoder      *admission.Decoder
+	NpController controllers.NetworkPolicyReconciler
+}
+
+// Handle handles admission request for ConfigMap
+func (v *ConfigMapValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	v.Log.V(1).Info("Received admission webhook request", "Name", req.Name, "Operation", req.Operation)
+	switch req.Operation {
+	case admissionv1.Create:
+		return v.validateCreate()
+	case admissionv1.Update:
+		return v.validateUpdate(req)
+	case admissionv1.Delete:
+		return v.validateDelete()
+	default:
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf("invalid admission webhook request"))
+	}
+}
+
+// InjectDecoder injects the decoder.
+func (v *ConfigMapValidator) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
+
+// validateCreate validates the ConfigMap create
+func (v *ConfigMapValidator) validateCreate() admission.Response {
+	return admission.Allowed("")
+}
+
+// validateUpdate validates the ConfigMap and CloudResourcePrefix update
+func (v *ConfigMapValidator) validateUpdate(req admission.Request) admission.Response {
+	newConfigMap := &corev1.ConfigMap{}
+	oldConfigMap := &corev1.ConfigMap{}
+	newControllerConfig := &config.ControllerConfig{}
+	oldControllerConfig := &config.ControllerConfig{}
+	err := v.decoder.Decode(req, newConfigMap)
+	if err != nil {
+		v.Log.Error(err, "failed to decode configMap", "ConfigMapValidator", req.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	if newConfigMap.GetName() != config.ConfigMap {
+		return admission.Allowed("")
+	}
+	newConfigData := newConfigMap.Data["nephe-controller.conf"]
+	err = yaml.UnmarshalStrict([]byte(newConfigData), newControllerConfig)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	err = json.Unmarshal(req.OldObject.Raw, &oldConfigMap)
+	if err != nil {
+		v.Log.Error(err, "failed to decode old configMap", "ConfigMapValidator", req.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	oldConfigData := oldConfigMap.Data["nephe-controller.conf"]
+	err = yaml.UnmarshalStrict([]byte(oldConfigData), oldControllerConfig)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	if newControllerConfig.CloudResourcePrefix == oldControllerConfig.CloudResourcePrefix {
+		v.Log.V(1).Info("CloudResourcePrefix are same", "newCloudResourcePrefix",
+			newControllerConfig.CloudResourcePrefix, "oldCloudResourcePrefix", oldControllerConfig.CloudResourcePrefix)
+		return admission.Allowed("")
```

Comment:
You should thrown an admission.Denied here, as nothing change from configuration perspective.

Reply:
If we add more variables in the nephe_controller-conf if one of the variables is modified then we hit some issue because that is considered as an update of the config map. Also, alex api throws an error.